cosmos/operators/local.py

```diff
@@ -450,7 +451,7 @@ def get_datasets(self, source: Literal["inputs", "outputs"]) -> list[Dataset]:
         uris = []
         for completed in self.openlineage_events_completes:
             for output in getattr(completed, source):
-                dataset_uri = output.namespace + "/" + output.name
+                dataset_uri = output.namespace + "/" + urllib.parse.quote(output.name)
```

Comment:
Can we please add a test case to demonstrate the benefits or the reason for leading to have this change?